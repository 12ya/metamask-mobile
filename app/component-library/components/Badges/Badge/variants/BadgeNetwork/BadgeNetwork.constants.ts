/* eslint-disable import/prefer-default-export */
import { SAMPLE_NETWORK_PROPS } from '../../../../Networks/Network/Network.constants';

// Internal dependencies.
import { BadgeNetworkProps } from './BadgeNetwork.types';

// Test IDs
export const BADGE_NETWORK_TEST_ID = 'badge-network';

// Sample consts
export const SAMPLE_BADGE_NETWORK_PROPS: BadgeNetworkProps = {
  networkProps: SAMPLE_NETWORK_PROPS,
};
