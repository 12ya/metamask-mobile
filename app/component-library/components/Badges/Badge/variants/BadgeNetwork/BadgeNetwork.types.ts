// External dependencies.
import { BadgeBaseProps } from '@component-library/components/Badges/Badge/foundation/BadgeBase/BadgeBase.types';
import { AvatarNetworkProps } from '@component-library/components/Avatars/Avatar/variants/AvatarNetwork/AvatarNetwork.types';

/**
 * BadgeNetwork component props.
 */
export interface BadgeNetworkProps
  extends Omit<BadgeBaseProps, 'children'>,
    AvatarNetworkProps {
  /**
   * Optional prop to control whether the Badge should be scaled to the content.
   * @default true
   */
  isScaled?: boolean;
}

/**
 * Style sheet BadgeNetwork parameters.
 */
export type BadgeNetworkStyleSheetVars = Pick<
  BadgeNetworkProps,
  'style' | 'size' | 'isScaled'
> & {
  containerSize: { width: number; height: number } | null;
};
