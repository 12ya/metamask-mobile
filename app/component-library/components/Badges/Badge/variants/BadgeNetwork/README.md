# BadgeNetwork

BadgeNetwork is a badge component that contains the Network as the content. **This component is not meant to be used by itself**. It is used by the [BadgeWrapper](../../../BadgeWrapper/BadgeWrapper.tsx) component, which can render this component as a badge.

## Props

This component extends [BadgeBaseProps](../../foundation/BadgeBase/BadgeBase.types.ts) from the [BadgeBase](../../foundation/BadgeBase/BadgeBase.tsx) component.

### `variant`

Variant of badge.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> |
| :-------------------------------------------------- | :------------------------------------------------------ |
| [BadgeVariants.Network](../../Badge.types.ts)                                           | No                                                     |

### `networkProps`

Props for the network content.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> |
| :-------------------------------------------------- | :------------------------------------------------------ |
| [NetworkProps](../../../../Networks/Network/Network.types.ts)                                              | Yes                                                     |


## Usage

```javascript
const networkImageProps = {
    source: {
        uri: SAMPLE_IMAGE_URI
    }
}
const networkProps = {
  size: NetworkSizes.Md,
  name: SAMPLE_NETWORK_NAME,
  imageProps: networkImageProps,
}

<BadgeNetwork networkProps={networkProps} />;
```
