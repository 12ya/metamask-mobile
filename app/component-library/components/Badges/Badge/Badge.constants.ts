/* eslint-disable import/prefer-default-export */

// External dependencies.
import { SAMPLE_BADGE_NETWORK_PROPS } from './variants/BadgeNetwork/BadgeNetwork.constants';

// Internal dependencies.
import { BadgeProps, BadgeVariants } from './Badge.types';

// Test IDs
export const BADGE_NETWORK_TEST_ID = 'badge-network';

// Sample consts
export const SAMPLE_BADGE_PROPS: BadgeProps = {
  variant: BadgeVariants.Network,
  ...SAMPLE_BADGE_NETWORK_PROPS,
};
