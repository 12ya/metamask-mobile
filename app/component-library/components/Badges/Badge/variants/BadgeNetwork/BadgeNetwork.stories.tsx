// Third party dependencies.
import React from 'react';

// External dependencies.
import { NetworkProps } from '../../../../Networks/Network/Network.types';
import { getNetworkStoryProps } from '../../../../Networks/Network/Network.stories';

// Internal dependencies.
import BadgeNetwork from './BadgeNetwork';
import { BadgeNetworkProps } from './BadgeNetwork.types';

export const getBadgeNetworkStoryProps = (): BadgeNetworkProps => {
  const networkProps: NetworkProps = getNetworkStoryProps();

  return {
    networkProps,
  };
};
const BadgeNetworkStory = () => (
  <BadgeNetwork {...getBadgeNetworkStoryProps()} />
);

export default BadgeNetworkStory;
