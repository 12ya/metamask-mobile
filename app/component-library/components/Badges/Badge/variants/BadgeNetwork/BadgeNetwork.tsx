/* eslint-disable react/prop-types */

// Third library dependencies.
import React from 'react';

// External dependencies.
import { useStyles } from '../../../../../hooks';
import BadgeBase from '../../foundation/BadgeBase';
import Network from '../../../../Networks/Network';

// Internal dependencies
import { BadgeNetworkProps } from './BadgeNetwork.types';
import styleSheet from './BadgeNetwork.styles';
import { BADGE_NETWORK_TEST_ID } from './BadgeNetwork.constants';

const BadgeNetwork = ({ networkProps, style }: BadgeNetworkProps) => {
  const { styles } = useStyles(styleSheet, { style });

  return (
    <BadgeBase style={styles.base} testID={BADGE_NETWORK_TEST_ID}>
      <Network {...networkProps} />
    </BadgeBase>
  );
};

export default BadgeNetwork;
