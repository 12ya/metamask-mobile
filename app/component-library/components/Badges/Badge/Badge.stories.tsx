/* eslint-disable no-console */
// Third party dependencies.
import React from 'react';
import { storiesOf } from '@storybook/react-native';

// External dependencies.
import { TEST_REMOTE_IMAGE_SOURCE } from './variants/BadgeNetwork/BadgeNetwork.constants';
import { BadgeVariants } from './Badge.types';

// Internal dependencies.
import Badge from './Badge';

storiesOf('Component Library / Badge', module).add('Default', () => (
  <Badge
    variant={BadgeVariants.Network}
    imageSource={TEST_REMOTE_IMAGE_SOURCE}
  />
));
