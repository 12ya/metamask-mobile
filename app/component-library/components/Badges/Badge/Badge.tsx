/* eslint-disable react/prop-types */
import React from 'react';

// External dependencies.
import BadgeAvatar from './variants/BadgeAvatar';

// Internal dependencies.
import { BadgeProps, BadgeVariants } from './Badge.types';
import { BADGE_AVATAR_TEST_ID } from './Badge.constants';

const Badge = (badgeProps: BadgeProps) => {
  switch (badgeProps.variant) {
    case BadgeVariants.Avatar:
      return <BadgeAvatar testID={BADGE_AVATAR_TEST_ID} {...badgeProps} />;
    default:
      throw new Error('Invalid Badge Variant');
  }
};

export default Badge;
