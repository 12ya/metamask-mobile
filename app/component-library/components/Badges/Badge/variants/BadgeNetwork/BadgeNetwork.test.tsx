// Third party dependencies.
import React from 'react';
import { shallow } from 'enzyme';

// External dependencies.
import { BadgeVariants } from '../../Badge.types';

// Internal dependencies.
import BadgeNetwork from './BadgeNetwork';
import {
  BADGE_NETWORK_TEST_ID,
  TEST_NETWORK_PROPS,
} from './BadgeNetwork.constants';

describe('BadgeNetwork - snapshots', () => {
  it('should render badge avatar', () => {
    const wrapper = shallow(
      <BadgeNetwork
        variant={BadgeVariants.Network}
        networkProps={TEST_NETWORK_PROPS}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('BadgeNetwork', () => {
  it('should render badge avatar with the given content', () => {
    const wrapper = shallow(
      <BadgeNetwork
        variant={BadgeVariants.Network}
        networkProps={TEST_NETWORK_PROPS}
      />,
    );

    const contentElement = wrapper.findWhere(
      (node) => node.prop('testID') === BADGE_NETWORK_TEST_ID,
    );
    expect(contentElement.exists()).toBe(true);
  });
});
