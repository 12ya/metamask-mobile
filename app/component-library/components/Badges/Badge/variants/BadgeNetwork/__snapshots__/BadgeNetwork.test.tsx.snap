// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeNetwork - snapshots should render badge avatar 1`] = `
<BadgeBase
  style={Object {}}
  testID="badge-network"
>
  <Network />
</BadgeBase>
`;
